### PR TITLE
test(processing): avoid JDK crash in schemaless segment test

### PR DESCRIPTION
**Pull request by FrankChen021**
## Summary

- Migrate `SchemalessTestSimpleTest` from class-level JUnit 4 parameterization to method-level JUnit 5 `@ParameterizedTest`.
- Create fresh incremental and merged fixtures for each method-source invocation.
- Use try-with-resources in each test method so every test invocation owns and closes its `Segment`.
- Close intermediate mapped indexes while retaining the returned merged index for the owning segment.

## JDK crash context

This change addresses the intermittent direct-buffer JVM crash observed in the crash reproduction PR [#20028](https://github.com/apache/druid/pull/20028), which affected `SchemalessTestSimpleTest` on JDK 25.

## Validation

- Processing test-compile passed.
- Focused test passed after the lifecycle change: 36 active executions; the existing disabled TopN test remains skipped.
- The equivalent lifecycle workaround was also exercised in 100 separate Maven invocations on Temurin JDK 25 with Surefire retries disabled, with no JDK crash.
- Checkstyle, SpotBugs, and git diff --check passed.

This PR is intended to validate whether explicit fixture ownership and cleanup prevent the intermittent JDK direct-buffer crash.

### CODE REVIEW

File: processing/src/test/java/org/apache/druid/segment/SchemalessIndexTest.java [hunk 1]
```diff
@@ -178,45 +183,62 @@
...
+    final IncrementalIndex top = makeIncrementalIndex("druid.sample.json.top", METRIC_AGGS);
+    final IncrementalIndex bottom = makeIncrementalIndex("druid.sample.json.bottom", METRIC_AGGS);
+    QueryableIndex topIndex = null;
+    QueryableIndex bottomIndex = null;
 
-        File topFile = new File(tmpFile, "top");
-        File bottomFile = new File(tmpFile, "bottom");
-        File mergedFile = new File(tmpFile, "merged");
+    try {
+      File tmpFile = File.createTempFile("yay", "who");
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply processing/src/test/java/org/apache/druid/segment/SchemalessIndexTest.java]: Addressed in commit `50c17d314f`. I replaced all five `File.createTempFile(...); delete(); mkdirp(...)` patterns in `SchemalessIndexTest` with Druid's `FileUtils.createTempDir(...)`, which uses `Files.createTempDirectory` to create the directory atomically with a unique name. This removes the predictable-path/re-created-file issue identified by CodeQL. Processing test-compile, the focused schemaless test (37 executions, 36 active), Checkstyle, SpotBugs, and `git diff --check` all pass.

**Review by FrankChen021 — COMMENTED** [review]
I have reviewed the code for correctness, edge cases, concurrency, and integration risks; no issues found.

Reviewed 2 of 2 changed files.

Validation: `git diff --check` passed.



---

This is an automated review by Codex GPT-5.6-Luna(max)
